### PR TITLE
ARROW-7580: [Website] 0.16 release post 

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

**Comment by pitrou** [comment]
We need to add a summary of the dataset additions. @fsaintjacques @bkietz ?

File: _posts/2020-01-25-0.16.0-release.md [hunk 1]
```diff
@@ -92,6 +92,33 @@ and result value, rather than taking a pointer-out function parameter
...
+`DatasetFactory` and `SourceFactory` interfaces. Notably `FileSystemSourceFactory`
+can crawl directories to find the candidates files (`Fragment`) supporting by a
+given `FileFormat`. The factories will also try to unify schemas, transparently
+supporting missing/added columns to the final unified schema. (ARROW-6614,
+ARROW-7061, ARROW-843, ARROW-7380)
+
+A partitioning facility was added to support partition pruning with predicate
+pushdown (ARROW-6494). The partitioning extracts partition from the `Fragment`'s
+path, e.g. `/data/year=2015/month=04/day=29` would extract the `year`, `month`
+and `day` partitions. Partitions can injected in the schema and the RecordBatch
```
Comment (bkietz) [inline _posts/2020-01-25-0.16.0-release.md]:
```suggestion
and `day` partitions. Partitions fields can injected into the schema and into the RecordBatch
```

File: _posts/2020-01-25-0.16.0-release.md [hunk 2]
```diff
@@ -92,6 +92,33 @@ and result value, rather than taking a pointer-out function parameter
...
+as materialized columns (ARROW-6965). Add a `PartitioningFactory` discovery
```
Comment (bkietz) [inline _posts/2020-01-25-0.16.0-release.md]:
```suggestion
as materialized columns (ARROW-6965). There is now a `PartitioningFactory` discovery
```

File: _posts/2020-01-25-0.16.0-release.md [hunk 3]
```diff
@@ -92,6 +92,33 @@ and result value, rather than taking a pointer-out function parameter
...
+facility such that types of the `Partitioning`'s schema are inferred if possible.
+
+The `ParquetFileFormat` transparently supports predicate pushdown by ignoring
+RowGroups based on their statistic (ARROW-6952). It also supports column
```
Comment (bkietz) [inline _posts/2020-01-25-0.16.0-release.md]:
```suggestion
RowGroups based on their statistics (ARROW-6952). It also supports column
```

File: _posts/2020-01-25-0.16.0-release.md [hunk 4]
```diff
@@ -92,6 +92,33 @@ and result value, rather than taking a pointer-out function parameter
...
+`IN` and `IS_VALID` filter operators (ARROW-7185).
+
+Classes were renamed to avoid the repeated `Data` prefix. The core classes are
+now `Dataset`, `Source`, `Fragment`, and `Partitioning` (ARROW-7498). The dataset
+APIs now use `Result<T>` when returning both a Status and result value, rather
+than taking a pointer-out function parameter (ARROW-7148).
+
+A discovery facility was added to create a `Dataset` and/or `Source` via the
+`DatasetFactory` and `SourceFactory` interfaces. Notably `FileSystemSourceFactory`
+can crawl directories to find the candidates files (`Fragment`) supporting by a
```
Comment (bkietz) [inline _posts/2020-01-25-0.16.0-release.md]:
```suggestion
can crawl directories to find the candidates files (`FileFragment`s) supported by a
```

**Comment by nevi-me** [comment]
Hi @nealrichardson, my sincere apologies; I was actually trying to push my commit on Rust changes into my fork, then open a PR. I think I had finger trouble. If you prefer, I can revert my commit and make it a PR instead.

@andygrove @paddyhoran @liurenjie1024 are you otherwise happy with the Rust notes additions? (Rendered: https://github.com/apache/arrow-site/pull/41/commits/74447c81cebbdc6f7164c48aa38981958234bc54?short_path=6283d65#diff-6283d65beb8b589c142c7e3d9c16ad64)

File: _posts/2020-01-25-0.16.0-release.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+* Fixed Size List and Fixed Size Binary
+* Adding a String Array for utf-8 strings, and keeping the Binary Array for general binary data
+* Duration and interval arrays.
+
+IPC readers for files and streams have been implemented (ARROW-5180).
+
+### Rust: DataFusion
+
+Query execution has been reimplemented with an extensible physical query plan. This allows other projec ts to add other plans, such as for distributed computing or for specific database servers (ARROW-5227).
```
Comment (paddyhoran) [inline _posts/2020-01-25-0.16.0-release.md]:
```suggestion
Query execution has been reimplemented with an extensible physical query plan. This allows other projects to add other plans, such as for distributed computing or for specific database servers (ARROW-5227).
```

**Comment by wesm** [comment]
I added some notes and compacted part of the C++ section. We might in general consider omitting most JIRA references to keep the information density as high as possible for readers who are less interested in the gory details.

File: _posts/2020-01-25-0.16.0-release.md [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+(ARROW-7420).
+
+## C# Notes
+
+
+## Java notes
+
+
+## Python notes
+
```
Comment (jorisvandenbossche) [inline _posts/2020-01-25-0.16.0-release.md]: Should we mention here that 0.16 is the last release to support Python 2.7 ?

Comment (wesm) [reply _posts/2020-01-25-0.16.0-release.md]: yes

**Comment by nealrichardson** [comment]
I removed the Jira references from the dataset section since it’s all new functionality. I agree that they aren’t interesting to probably anyone but us.

**Comment by kszucs** [comment]
@wesm @nealrichardson shall we merge this?

**Comment by nealrichardson** [comment]
Please no, not without reading through it first. There are still some placeholders that need to be filled in, e.g. https://github.com/apache/arrow-site/pull/41/files#diff-6283d65beb8b589c142c7e3d9c16ad64R28-R29

**Comment by wesm** [comment]
I think ample time has passed... I'm giving it a read through and will merge after that

**Comment by wesm** [comment]
I rebased and gave a skim. I'll leave this open for another half hour or so in case anyone wants to give it a skim and fix typos or reword things

**Pull request merged**